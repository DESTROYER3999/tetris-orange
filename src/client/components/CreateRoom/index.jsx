import { connect } from "react-redux";
import CreateRoomSub from "./subcomponent";

const mapStateToProps = state => ({
  playerName: state.player.name
});

const CreateRoom = connect(
  mapStateToProps,
  null
)(CreateRoomSub);

export default CreateRoom;
