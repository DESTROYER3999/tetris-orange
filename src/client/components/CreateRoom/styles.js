import styled from "styled-components";
import Input from "../common/Input";
import { withStyles } from "@material-ui/core/styles";

export const StyledForm = styled.form`
  align-self: center;
  width: 70%;
  display: flex;
  padding: 1vh;
  border-radius: 4px;
  justify-content: center;
  align-items: center;
  background: rgb(0, 0, 0, 0.2);
`;

export const StyledRoomNameContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledLabel = styled.label`
  padding: 0.5vh;
  font-size: 2vh;
  color: white;
`;

export const StyledButton = styled.button`
  height: 2vh;
  background: none;
  border: 0px;
  outline: none;
  opacity: 0;
  &:enabled {
    opacity: 1;
  }
  &:disabled {
    heigth: 0px;
  }
`;

export const StyledInput = withStyles({
  root: {
    "&": {
      marginTop: 0,
      marginRight: 10
    },
    "& label.Mui-focused": {
      color: "white"
    },
    "$ .MuiFormHelperText-contained p": {
      height: "2vh"
    },
    "& label": {
      color: "white"
    },
    "& input": {
      color: "white"
    },
    "& .MuiFormHelperText-root": {
      backgroundColor: "white",
      fontSize: "1.5vh"
    },
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "white"
      },
      "&:hover fieldset": {
        borderColor: "white"
      },
      "&.Mui-focused fieldset": {
        borderColor: "white"
      },
      "&.Mui-error fieldset": {
        borderColor: "white"
      }
    }
  }
})(Input);
