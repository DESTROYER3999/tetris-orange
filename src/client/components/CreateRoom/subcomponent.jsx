import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import {
  StyledForm,
  StyledRoomNameContainer,
  StyledLabel,
  StyledButton,
  StyledInput
} from "./styles";
import inputError from "../../errors/inputError";

const CreateRoomSub = ({ playerName }) => {
  const [error, setError] = useState({ boolean: null, message: "" });
  const [roomName, setRoomName] = useState("");

  const handleSubmit = e => {
    e.preventDefault();
    window.location.hash = roomName + "[" + playerName + "]";
  };

  const handleRoomNameChange = e => {
    const value = e.target.value;
    setError(inputError(value));
    setRoomName(value);
  };

  useEffect(() => {
    document.getElementById("name").focus();
  });

  return (
    <StyledForm id="form" onSubmit={e => handleSubmit(e)}>
      <StyledRoomNameContainer>
        <StyledLabel htmlFor="name">Create Room</StyledLabel>
        <StyledInput
          id="name"
          error={error.boolean}
          helperText={error.message}
          spellCheck="false"
          autoComplete="off"
          value={roomName}
          onChange={handleRoomNameChange}
        />
        <StyledButton id="enter" disabled={!roomName || error.boolean}>
          <p>Press Enter</p>
        </StyledButton>
      </StyledRoomNameContainer>
    </StyledForm>
  );
};

CreateRoomSub.propTypes = {
  playerName: PropTypes.string.isRequired
};

export default CreateRoomSub;
