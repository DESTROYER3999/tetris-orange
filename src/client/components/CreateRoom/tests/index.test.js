import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import CreateRoom from "../index";

const mockStore = configureStore();
const initialState = {
  player: { name: "Bob" }
};
const store = mockStore(initialState);

describe("<CreateRoom />", () => {
  test("CreateRoom rendering with store without crashing", () => {
    const wrapper = shallow(<CreateRoom store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
