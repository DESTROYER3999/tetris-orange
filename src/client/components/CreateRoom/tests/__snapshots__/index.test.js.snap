// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoom /> CreateRoom rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <CreateRoomSub
    dispatch={[Function]}
    playerName="Bob"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
